src/routes/products.ts

```diff
@@ -0,0 +1,25 @@
+import { Router, Request, Response } from "express";
+import {
+	getProducts,
+	getProduct,
+	addProduct,
+	updateProduct,
+	deleteProduct
+} from "../controllers/ProductsController";
+
+import { productValidation } from "../utils/validations";
+import handleValidationErrors from "../utils/handleValidationErrors";
+
+const productsRoter: Router = Router();
+
+productsRoter.get('/', getProducts)
+
+productsRoter.get('/:id', getProduct)
+
+productsRoter.post('/', productValidation, handleValidationErrors, addProduct)
+
+productsRoter.put('/:id', updateProduct)
```

Comment:
i think this should be patch request, as it only updates the manufacturer.address.street field, also I would change the name of the function to be more descriptive. e.g updadeProductAddress